src/wallet/wallet.h

```diff
@@ -95,7 +95,7 @@ enum WalletFeature
 
     FEATURE_NO_DEFAULT_KEY = 159900, // Wallet without a default key written
 
-    FEATURE_LATEST = FEATURE_COMPRPUBKEY // HD is optional, use FEATURE_COMPRPUBKEY as latest version
+    FEATURE_LATEST = FEATURE_NO_DEFAULT_KEY
```

Comment:
I dont think we should "upgrade" to NO_DEFAULT_KEY (unless we delete the default key, but that seems useless). Rather, we should upgrade to HD_SPLIT and then upgrade to NO_DEFAULT_KEY when we next bump FEATURE_LATEST (probably leave a comment to that effect here).

Reply:
I don't see why not upgrade to NO_DEFAULT_KEY. FEATURE_LATEST is only used for new wallets and upgraded wallets; users should expect new and upgraded wallets to not be backwards compatible.

Reply:
NO_DEFAULT_KEY doesnt even have a meaning in this case, I mean you're upgrading to nVersion == NO_DEFAULT_KEY but you still have a default key? Its just needless incompatibility at that point (though, I agree, not a huge deal given the user asked for that).

Reply:
@TheBlueMatt Does that matter? This is just the default for new wallets or explicit upgrades. I don't really care about gratuitous incompatibility when you're explicitly not caring about that.

UPDATE: especially with the new PRE_SPLIT type added afterwards this is probably inevitable.